### PR TITLE
Updates Base Goodie Re: #2767

### PR DESCRIPTION
**Pull request by Mylainos**
**Issue #2767**
Upgrade the IA to full template.

---
IA page: https://duck.co/ia/view/base
Maintainer: @moritz

### CODE REVIEW

**Comment by daxtheduck** [comment]
@Mylainos Hey!
        
Thanks for taking the time to contribute! We really appreciate it.

We work closely with every contributor to make Instant Answers the best they can be, so we appreciate your patience as we look over your code. From here, the process usually goes like this:

1. Pull Request is reviewed by the DuckDuckGo staff and community
2. Staff and community will leave feedback with any necessary updates to the function or design.
3. Once you've made any necessary corrections, then your Instant Answer will be merged and deployed live on DuckDuckGo!

If you have any questions along the way, feel free to ask them here. Our staff and community are also available on Slack to answer any questions you may have. If you'd like to join us there please send an email to <quackslack@duckduckgo.com>.

>More Info: http://docs.duckduckhack.com/submitting/submitting-overview.html

Thanks!

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

File: t/Base.t [hunk 1]
```diff
@@ -8,91 +8,34 @@ use DDG::Test::Goodie;
...
+ddg_goodie_test(
+    [qw(DDG::Goodie::Base)],
+
+    '255 in hex' => build_test(255, 16, 'FF'),
+    '255 in base 16' => build_test(255, 16, 'FF'),
+    '42 in binary' => build_test(42, 2, '101010'),
+    '42 in base 2' => build_test(42, 2, '101010'),
+    '42 to hex' => build_test(42, 16, '2A'),
+    '42 to octal' => build_test(42, 8, '52'),
+    '10 in base 3' => build_test(10, 3, '101'),
```
Comment (GuiltyDolphin) [inline t/Base.t]: Mind aligning these? They look so pretty when aligned... :blush:

Comment (mintsoft) [inline t/Base.t]: @GuiltyDolphin that sounds so creepy...

File: t/Base.t [hunk 2]
```diff
@@ -8,91 +8,34 @@ use DDG::Test::Goodie;
...
+
+    '255 in hex'     => build_test(255, 16, 'FF'),
+    '255 in base 16' => build_test(255, 16, 'FF'),
+    '42 in binary'   => build_test(42, 2, '101010'),
+    '42 in base 2'   => build_test(42, 2, '101010'),
+    '42 to hex'      => build_test(42, 16, '2A'),
+    '42 to octal'    => build_test(42, 8, '52'),
+    '10 in base 3'   => build_test(10, 3, '101'),
+    '18442240474082181119 to hex'     => build_test(18442240474082181119, 16, 'FFEFFFFFFFFFFFFF'),
+    '999999999999999999999999 to hex' => build_test("999999999999999999999999", 16, 'D3C21BCECCEDA0FFFFFF'),
```
Comment (Mylainos) [inline t/Base.t]: Want more alignment?

File: t/Base.t (hunk 1 above)
Comment (GuiltyDolphin) [inline t/Base.t]: Good ole `vi(:Tabularize /=>`.

File: t/Base.t (hunk 2 above)
Comment (GuiltyDolphin) [inline t/Base.t]: Oh boy, do I? (Yes).

Comment (Mylainos) [inline t/Base.t]: So this is good ?

Comment (mintsoft) [inline t/Base.t]: Yeah he'll be fine with that

File: t/Base.t [hunk 3]
```diff
@@ -8,91 +8,34 @@ use DDG::Test::Goodie;
...
-          DDG::Goodie::Base
-          )
-    ],
+sub build_structured_answer {
+    my ($number, $base, $based) = @_;
+    return "$number in base $base is $based",
+        structured_answer => {
+            data => {
+                title => $based,
+                subtitle => "Decimal to base $base: $number"
```
Comment (GuiltyDolphin) [inline t/Base.t]: If you're in the mood for aligning.... :sweat_smile:

**Comment by Mylainos** [comment]
Done

**Comment by GuiltyDolphin** [comment]
@Mylainos Thank yea! :dolphin:

**Pull request merged**